build.gradle

```diff
@@ -202,6 +202,22 @@ model {
     }
 }
 
+project(':devApp') {
+    apply plugin: 'java'
+    apply plugin: 'application'
+
+    dependencies {
+        compile project(':')
+        runtime files(project(':').nativeTestFilesJar.archivePath)
+    }
+
+    sourceSets.main.java.srcDirs = [ "${rootDir}/src/dev/java" ]
```

Comment:
You can create a separate jar from a specific source set. You need to define a JAR task that pulls from that source set. You may be able to configure the application plugin to use that source set instead of the default `src` one.

This is why I say it's hacky. Given this is a gradle project you should work with it and not fight it like this. But you also shouldn't put another project's source code under its parent's `src` directory.